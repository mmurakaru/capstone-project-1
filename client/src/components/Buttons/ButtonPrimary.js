import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

const ButtonStyled = styled.button`
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  color: var(--primary-dark);
  font-size: 0.8em;
  padding: 6px;
  width: 100%;

  &:disabled {
    opacity: 50%;
  }
`
const ButtonPrimary = (props) => <ButtonStyled {...props} />

ButtonPrimary.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  children: PropTypes.string.isRequired,
}

export default ButtonPrimary
