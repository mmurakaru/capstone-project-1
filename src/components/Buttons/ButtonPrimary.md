Regular Button with text

```js
<ButtonPrimary disabled={false} onClick={() => {}}>
  Save this song
</ButtonPrimary>
```

```js
<ButtonPrimary disabled={true} onClick={() => {}}>
  Save this song
</ButtonPrimary>
```
