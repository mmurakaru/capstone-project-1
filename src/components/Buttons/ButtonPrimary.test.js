import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import ButtonPrimary from './ButtonPrimary'

describe('ButtonPrimary', () => {
  it('renders correctly', () => {
    const { container, rerender } = render(
      <ButtonPrimary disabled={false} onClick={() => {}}>
        Save this song
      </ButtonPrimary>
    )
    expect(container.firstChild).toMatchSnapshot()

    rerender(
      <ButtonPrimary disabled={true} onClick={() => {}}>
        Save this song
      </ButtonPrimary>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('calls onClick with the correct id', () => {
    const onClickMock = jest.fn()
    const { getByRole } = render(
      <ButtonPrimary disabled={false} onClick={() => onClickMock('a')}>
        Save this song
      </ButtonPrimary>
    )
    const button = getByRole('button')
    user.click(button)

    expect(onClickMock).toHaveBeenCalledWith('a')
  })
})
