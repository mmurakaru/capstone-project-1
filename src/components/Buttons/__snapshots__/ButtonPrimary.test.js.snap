// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimary renders correctly 1`] = `
.c0 {
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  color: var(--primary-dark);
  font-size: 0.8em;
  padding: 6px;
  width: 100%;
}

.c0:disabled {
  opacity: 50%;
}

<button
  class="c0"
>
  Save this song
</button>
`;

exports[`ButtonPrimary renders correctly 2`] = `
<button
  class="ButtonPrimary__ButtonStyled-sc-1mu6zlu-0 xEHRn"
  disabled=""
>
  Save this song
</button>
`;
